lib/state_machine/src/state_machine.rs

```diff
@@ -0,0 +1,59 @@
+use crate::types::State;
+use defmt::{info, warn};
+use embassy_net::tcp::TcpSocket;
+use hyped_core::mqtt::HypedMqttClient;
+use rust_mqtt::utils::rng_generator::CountingRng;
+
+pub struct StateMachine<'a> {
+    pub(crate) current_state: State,
+    pub(crate) mqtt_client: HypedMqttClient<'a, TcpSocket<'a>, CountingRng>,
```

Comment:
why are these `pub(crate)`? surely you don't want people to set the state randomly without going through `handle_transition(...)`?